engines/agos/metaengine.cpp

```diff
@@ -207,6 +213,158 @@ SaveStateList AgosMetaEngine::listSaves(const char *target) const {
+	act->addDefaultInputMapping("C+f");
+	engineKeyMap->addAction(act);
+
+	if (gameId == "waxworks" ||
+		gameId == "elvira1" ||
+		gameId == "elvira2" ||
+		gameId == "swampy" ||
+		gameId == "puzzle" ||
+		gameId == "jumble" ||
+		gameId == "dimp") {
+		act = new Action("WLKFORWARD", _("Walk forward")); // KEYCODE_UP
+		act->setCustomEngineActionEvent(AGOSAction::kActionWalkForward);
+		act->addDefaultInputMapping("UP"); 
+		act->addDefaultInputMapping("JOY_UP"); 
+		gameKeyMap->addAction(act);
+
+		act = new Action("TRNBACK", _("Turn backward")); // KEYCODE_DOWN
+		act->setCustomEngineActionEvent(AGOSAction::kActionTurnBack);
+		act->addDefaultInputMapping("DOWN"); 
+		act->addDefaultInputMapping("JOY_DOWN"); 
+		gameKeyMap->addAction(act);
+
+		act = new Action("TRNLEFT", _("Turn left")); // KEYCODE_LEFT
+		act->setCustomEngineActionEvent(AGOSAction::kActionTurnLeft);
+		act->addDefaultInputMapping("LEFT"); 
+		act->addDefaultInputMapping("JOY_LEFT"); 
+		gameKeyMap->addAction(act);
+
+		act = new Action("TRNRIGHT", _("Turn right")); // KEYCODE_RIGHT
+		act->setCustomEngineActionEvent(AGOSAction::kActionTurnRight);
+		act->addDefaultInputMapping("RIGHT"); 
+		act->addDefaultInputMapping("JOY_RIGHT"); 
+		gameKeyMap->addAction(act);
+	}
+
+	if (gameId == "simon1" ||
+		gameId == "simon2") {
+
+		act = new Action("TXTSDPUP_1", _("Text speed - Fast"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionTextSpeedFast);
+		act->addDefaultInputMapping("F1");
+		gameKeyMap->addAction(act);
+
+		act = new Action("TXTSDPUP_2", _("Text speed - Medium"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionTextSpeedMedium);
+		act->addDefaultInputMapping("F2");
+		gameKeyMap->addAction(act);
+
+		act = new Action("TXTSDPUP_3", _("Text speed - Slow"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionTextSpeedSlow);
+		act->addDefaultInputMapping("F3");
+		gameKeyMap->addAction(act);
+
+		act = new Action("SHOWOBJINTERACT", _("Show objects to interact"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionShowObjects);
+		act->addDefaultInputMapping("F10");
+		gameKeyMap->addAction(act);
+
+		if (gameId == "simon2") {
+			act = new Action("BACKGRNDSND", _("Toggle background sounds On/Off"));
+			act->setCustomEngineActionEvent(AGOSAction::kActionToggleBackgroundSound);
+			act->addDefaultInputMapping("b");
+			gameKeyMap->addAction(act);
+		}
+	}
+
+	if (gameId == "feeble") {
+		act = new Action("SWTCHCHARACTER", _("Switch characters"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionToggleSwitchCharacter);
+		act->addDefaultInputMapping("F7");
+		gameKeyMap->addAction(act);
+
+		act = new Action("TOGGLEHITBOX", _("Toggle hitbox names on/off"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionToggleHitboxName);
+		act->addDefaultInputMapping("F9");
+		gameKeyMap->addAction(act);
+	}
+
+	if (gameId == "feeble" ||
+		gameId == "simon2") {
+		act = new Action("TOGGLESPEECH_SUB", _("Switches between speech and subtitles"));
```

Comment:
Is this description right? The handler looks like it should be "Switches between speech only and combined speech and subtitles"

Reply:
Yes, you are right, i copied word by word from the game manual thus I had the confusion with enum names too, fixing it now